compiler/z/codegen/UnaryEvaluator.cpp

```diff
@@ -520,9 +520,14 @@ OMR::Z::TreeEvaluator::l2aEvaluator(TR::Node * node, TR::CodeGenerator * cg)
    node->setRegister(source);
 
 
-   //printf("Node %p is first child\n", firstChild); fflush(stdout);
-   //traceMsg(comp, "Node %p is first child\n", firstChild);
-   cg->decReferenceCount(firstChild);
+   if (shouldGenerateGuardedLoadsForCompressedLoad)
+      {
+      cg->decReferenceCount(firstChild->getFirstChild());
```

Comment:
In light of your proposed fix, who now decrements the ref count on `firstChild` in this case?  For that matter, who evaluates the first child when `shouldGenerateGuardedLoadsForCompressedLoad` is true?